sn_networking/src/record_store.rs

```diff
@@ -210,6 +211,33 @@ impl DiskBackedRecordStore {
             warn!("Record storage didn't have the distance_range setup yet.");
         }
     }
+
+    #[allow(dead_code)]
+    /// Calculate the cost to store data for our current store state
+    pub fn store_cost(&self) -> Token {
+        let records_len = self.records().count();
+
+        let mut cost = Token::from_nano(1);
+        const NANOS_PER_SNT: u64 = 1_000_000_000;
+
+        // if we imagine that at FULL capacity we want _all the money_ (ie, we never want to reach full capacity)
+        // So we get the total amount of nanos in the network,
+        // we divide that number by our MAX_RECORD_COUNT to know
+        // what to charge for each additional record
+        let step_per_record = ((2 ^ 32) * NANOS_PER_SNT) / MAX_RECORDS_COUNT as u64;
+        let step_amount = Token::from_nano(step_per_record);
+
+        // we want to spread the cost of storing data across the MAX_CAPACITY
+        // doubling the cost for each record store
+        for _ in 0..records_len {
+            if let Some(new_cost) = cost.checked_add(step_amount) {
```

Comment:
I don't get this.

first, what's the diff to just using : `cost.saturating_add(records_len * step_amount)`

second, the `step_amount` is just a `cost_per_record`. 
However, as I understanding, the `stepping charge` shall be something like:
For 1-10 records, charge 1 per record,
For 11-20 records, charge 2 per record,
For 21-40 records, charge 3 per record,
......
But here, seems the cost will be change `for each record`, which seems like more aggressive?

Reply:
>  Also, this will make all nodes (storing the same record) giving more various cost charge of that record, as they may holding different set of records due to different space address or lifespan in the network.
We can expect this anyway. Records stored vary in the hundreds sometimes. (at least on once machine). So we'll need to deal with that. (Which the plan is to just average the majority or responses. and pay _that_).

>But here, seems the cost will be change for each record,
Ahhh, yeh here I'm missing any module calc like the pruning!